src/components/datepicker/DatePickerInput.js

```diff
@@ -36,8 +36,11 @@ class DatePickerInput extends PureComponent {
     this.setState({ inputHasFocus: true }, () => this.props.onFocus && this.props.onFocus());
   };
 
-  handleInputDateChange = date => {
-    this.setState({ selectedDate: date }, () => this.props.onChange(date));
+  handleInputDateChange = (date, modifiers = {}) => {
+    if (modifiers.disabled) {
```

Comment:
Q: Why not just an inverse check and put it in the if?

Reply:
I like to keep my edge cases separate, and I suppose there can be other modifiers than can influence the date being called or not as well.